src/main/java/org/jraf/androidcontentprovidergenerator/Main.java

```diff
@@ -128,7 +124,7 @@ public boolean accept(File pathname) {
                 if (fieldDocumentation.isEmpty()) fieldDocumentation = null;
                 String type = fieldJson.getString(Field.Json.TYPE);
                 boolean isIndex = fieldJson.optBoolean(Field.Json.INDEX, false);
-                boolean isNullable = fieldJson.optBoolean(Field.Json.NULLABLE, true);
+                boolean isNullable = !isIndex && fieldJson.optBoolean(Field.Json.NULLABLE, true);
```

Comment:
Having an index an a null value sound weird to me (ie a non existant value). I don't know if SQlite allows it, but why not.